test/boost/sstable_resharding_test.cc

```diff
@@ -30,11 +31,12 @@ static inline std::vector<std::pair<sstring, dht::token>> token_generation_for_s
 
     key_and_token_pair.reserve(tokens_to_generate);
     dht::set_global_partitioner(to_sstring("org.apache.cassandra.dht.Murmur3Partitioner"));
+    dht::murmur3_partitioner partitioner(smp::count, 0);
```

Comment:
If we have dht::set_global_partitioner(), is there also a dht::get_global_partitioner() so you don't need to repeat the same partitioner definition you defined right above it?

Reply:
The goal of this PR is to remove global partitioner. All the commits are steps towards that direction. Here I change the test to create a partitioner it needs and use them directly instead of setting the global partitioner and use it implicitly.